src/boot_O2/loadfragment.c

```diff
@@ -1,8 +1,171 @@
+                 */
+                luiRefs[(*relocDataP >> 0x10) & 0x1F] = relocDataP;
+                luiVals[(*relocDataP >> 0x10) & 0x1F] = *relocDataP;
+                break;
+            case 0x6000000:
+                /* R_MIPS_LO16
+                 * Updates the LUI instruction to reflect the relocated address.
+                 * The full address is calculated from the LUI and lo parts, and then updated.
+                 * if the lo part is negative, add 1 to the lui.
+                 */
+                luiInstRef = luiRefs[(*relocDataP >> 0x15) & 0x1F];
+                regValP = &luiVals[((*relocDataP) >> 0x15) & 0x1F];
+                if ((((*luiInstRef << 0x10) + (s16)*relocDataP) & 0x0F000000) == 0) {
+                    relocatedAddress = (*regValP << 0x10) + (s16)*relocDataP - vRamStart + allocu32;
+                    isLoNeg = (relocatedAddress & 0x8000) ? 1 : 0;
+                    *luiInstRef = (*luiInstRef & 0xFFFF0000) | (((relocatedAddress >> 0x10) & 0xFFFF) + isLoNeg);
+                    *relocDataP = (*relocDataP & 0xFFFF0000) | (relocatedAddress & 0xFFFF);
+                } else {
+                    if (gLoadLogSeverity >= 3) {}
+                }
+                break;
+        }
+    }
+}
+
+size_t Load_LoadOverlay(u32 vRomStart, u32 vRomEnd, u32 vRamStart, void* allocatedVRamAddr, size_t allocatedBytes) {
+    size_t size = vRomEnd - vRomStart;
+    void* end;
+    s32 pad;
+    OverlayRelocationSection* ovl;
+
+    if (gLoadLogSeverity >= 3) {}
+    if (gLoadLogSeverity >= 3) {}
+
+    end = (uintptr_t)allocatedVRamAddr + size;
+    DmaMgr_SendRequest0(allocatedVRamAddr, vRomStart, size);
+
+    ovl = (OverlayRelocationSection*)((uintptr_t)end - ((s32*)end)[-1]);
+
+    if (gLoadLogSeverity >= 3) {}
+
+    if (allocatedBytes < ovl->bssSize + size) {
+        if (gLoadLogSeverity >= 3) {}
+        return 0;
+    }
+
+    allocatedBytes = ovl->bssSize + size;
+
+    if (gLoadLogSeverity >= 3) {}
+
+    Load_Relocate(allocatedVRamAddr, ovl, vRamStart);
+
+    if (ovl->bssSize != 0) {
+        if (gLoadLogSeverity >= 3) {}
+        bzero(end, ovl->bssSize);
+    }
+
+    osWritebackDCache(allocatedVRamAddr, allocatedBytes);
+    osInvalICache(allocatedVRamAddr, allocatedBytes);
+
+    if (gLoadLogSeverity >= 3) {}
+
+    return allocatedBytes;
+}
+
+void* Load_AllocateAndLoad(u32 vRomStart, u32 vRomEnd, u32 vRamStart) {
+    size_t size = vRomEnd - vRomStart;
+    void* end;
+    void* allocatedVRamAddr;
+    uintptr_t ovlOffset;
+    OverlayRelocationSection* ovl;
+    size_t bssSize;
+
+    if (gLoadLogSeverity >= 3) {}
+
+    allocatedVRamAddr = SystemArena_MallocR(size);
+    end = (uintptr_t)allocatedVRamAddr + size;
+
+    if (gLoadLogSeverity >= 3) {}
+
+    DmaMgr_SendRequest0(allocatedVRamAddr, vRomStart, size);
+
+    if (gLoadLogSeverity >= 3) {}
+
+    ovlOffset = (uintptr_t)end - 4;
+    ovl = (OverlayRelocationSection*)((uintptr_t)end - ((s32*)end)[-1]);
+
+    if (1) {}
```

Comment:
Just noting here. This actually does not match as `if(gLoadLogSeverity >= 3){}` so feels like a fake match. Based on jp 1.1 I think there should be a debug statement here, but its hard to tell with just the raw m2c output.